boefjes/boefjes/clients/scheduler_client.py

```diff
@@ -72,14 +81,20 @@ def get_queues(self) -> list[Queue]:
 
         return TypeAdapter(list[Queue]).validate_json(response.content)
 
-    def pop_item(self, queue_id: str) -> Task | None:
-        response = self._session.post(f"/queues/{queue_id}/pop")
+    def pop_item(self, scheduler_id: str) -> PaginatedTasksResponse | None:
+        response = self._session.post(f"/schedulers/{scheduler_id}/pop?limit=1")
+        self._verify_response(response)
+
+        return TypeAdapter(PaginatedTasksResponse | None).validate_json(response.content)
+
+    def pop_items(self, scheduler_id: str, filters: dict[str, Any]) -> PaginatedTasksResponse | None:
+        response = self._session.post(f"/schedulers/{scheduler_id}/pop", json=filters)
```

Comment:
Important to note is that the `pop` endpoint changed to return multiple `Task`s, additionally it allows for more complex filtering options. I've, for now limited the response to 1 `Task` to resemble how it currently works, but this can be updated accordingly to what is desired.